buildSrc/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -163,15 +163,29 @@ private static void setupDownloadServiceRepo(Project project) {
             });
             ivyRepo.getAuthentication().create("header", HttpHeaderAuthentication.class);
             ivyRepo.patternLayout(layout -> layout.artifact("/downloads/elasticsearch/[module]-[revision](-[classifier]).[ext]"));
-            ivyRepo.content(content -> content.includeGroup(FAKE_IVY_GROUP));
+            ivyRepo.content(content -> content.includeGroup(group));
         });
+    }
+
+    private static void setupDownloadServiceRepo(Project project) {
+        if (project.getRepositories().findByName(DOWNLOAD_REPO_NAME) != null) {
+            return;
+        }
+        addIvyRepo(project, DOWNLOAD_REPO_NAME, "https://artifacts.elastic.co", FAKE_IVY_GROUP);
+        if (ClasspathUtils.isElasticsearchProject(project) == false) {
+            // external, so add snapshot repo as well
+            addIvyRepo(project, SNAPSHOT_REPO_NAME, "https://snapshots.elastic.co", FAKE_SNAPSHOT_IVY_GROUP);
+        }
         project.getRepositories().all(repo -> {
             if (repo.getName().equals(DOWNLOAD_REPO_NAME) == false) {
                 // all other repos should ignore the special group name
                 repo.content(content -> content.excludeGroup(FAKE_IVY_GROUP));
             }
+            if (repo.getName().equals(SNAPSHOT_REPO_NAME) == false) {
```

Comment:
It might make sense to add the registration of this ignore rule into the `addIvyRepo()` method itself so we don't have to hard-code these for every repo we add.